sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,780 @@
+
+    out = cat((x_, zero_pad), axis=axis)
+    if isinstance(x, np.ndarray):
+        out = np.array(out)
+    return out
+
+
+def _create_mask(ones_length, zeros_length=0):
+    from torch import cat, ones, zeros
+
+    zeros_tensor = zeros(zeros_length)
+    ones_tensor = ones(ones_length)
+
+    input_mask = cat((ones_tensor, zeros_tensor))
+    return input_mask
+
+
+def _run_epoch(
+    cur_epoch,
+    accelerator,
+    criterion,
+    optimizer,
+    scheduler,
+    model,
+    max_norm,
+    train_dataloader,
+    val_dataloader,
+):
+    import torch.cuda.amp
+    from tqdm import tqdm
+
+    from sktime.libs.momentfm.utils.forecasting_metrics import get_forecasting_metrics
+
+    losses = []
+    for data in tqdm(train_dataloader, total=len(train_dataloader)):
+        # Move the data to the GPU
+        timeseries = data["historical_y"]
+        input_mask = data["input_mask"]
+        forecast = data["future_y"]
+        with torch.cuda.amp.autocast():
+            output = model(x_enc=timeseries, mask=input_mask)
+        loss = criterion(output.forecast, forecast)
+
+        accelerator.backward(loss)
+
+        # Clip gradients
+        torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm)
+
+        optimizer.step()
+        optimizer.zero_grad(set_to_none=True)
+
+        losses.append(loss.item())
+
+    losses = np.array(losses)
+    average_loss = np.average(losses)
+    tqdm.write(f"Epoch {cur_epoch}: Train loss: {average_loss:.3f}")
+
+    # Step the learning rate scheduler
+    scheduler.step()
+
+    # Evaluate the model on the test split
+    if val_dataloader:
+        trues, preds, histories, losses = [], [], [], []
+        model.eval()
+        with torch.no_grad():
+            for data in tqdm(val_dataloader, total=len(val_dataloader)):
+                # Move the data to the specified device
+                timeseries = data["historical_y"]
+                input_mask = data["input_mask"]
+                forecast = data["future_y"]
+
+                with torch.cuda.amp.autocast():
+                    output = model(x_enc=timeseries, mask=input_mask)
+
+                loss = criterion(output.forecast, forecast)
+                losses.append(loss.item())
+
+                trues.append(forecast.detach().cpu().numpy())
+                preds.append(output.forecast.detach().cpu().numpy())
+                histories.append(timeseries.detach().cpu().numpy())
+
+        losses = np.array(losses)
+        average_loss = np.average(losses)
+        model.train()
+
+        trues = np.concatenate(trues, axis=0)
+        preds = np.concatenate(preds, axis=0)
+        histories = np.concatenate(histories, axis=0)
+
+        metrics = get_forecasting_metrics(y=trues, y_hat=preds, reduction="mean")
+        tqdm.write(
+            f"Epoch {cur_epoch}: Test MSE: {metrics.mse:.3f}"
+            f"Test MAE: {metrics.mae:.3f}"
+        )
+    cur_epoch += 1
+    return cur_epoch
+
+
+def _check_device(device):
```

Comment:
Can you check if such code / method is directly available from torch, I would suppose it is.

Reply:
I don't believe there is, but we could remove it all together and let `accelerate` handle all of it as it checks the device automatically